notebooks/opensearch-writer.ipynb

```diff
@@ -0,0 +1,374 @@
+   "source": [
+    "import os\n",
+    "aryn_api_key = os.getenv(\"ARYN_API_KEY\") ## REPLACE WITH YOUR-ARYN-API-KEY"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "paths = [\"s3://aryn-public/NTSB/\"]\n",
+    "fsys = pyarrow.fs.S3FileSystem(region=\"us-east-1\", anonymous=True)\n",
+    "\n",
+    "llm = OpenAI(OpenAIModels.GPT_3_5_TURBO.value)\n",
+    "tokenizer = HuggingFaceTokenizer(\"thenlper/gte-small\")\n",
+    "\n",
+    "ctx = sycamore.init()\n",
+    "\n",
+    "ds = (\n",
+    "    ctx.read.binary(paths, binary_format=\"pdf\", filesystem=fsys)\n",
+    "    # Partition with the Aryn partitioner remotely, pulling out tables and images.\n",
+    "    .partition(partitioner=ArynPartitioner(aryn_api_key=aryn_api_key, use_partitioning_service=True, use_cache=False, extract_images=True,  extract_table_structure=True))\n",
+    "    # Get rid of spurious whitespace charaters\n",
+    "    .regex_replace(COALESCE_WHITESPACE)\n",
+    "    # Automatically determine a schema of additional metadata to extract from Documents\n",
+    "    .extract_batch_schema(schema_extractor=OpenAISchemaExtractor(\"FlightAccidentReport\", llm=llm, num_of_elements=35))\n",
+    "    # Extract the metadata specified by that schema\n",
+    "    .extract_properties(property_extractor=OpenAIPropertyExtractor(llm=llm, num_of_elements=35))\n",
+    "    # Merge elements into larger chunks\n",
+    "    .mark_bbox_preset(tokenizer=tokenizer)\n",
+    "    .merge(merger=MarkedMerger())\n",
+    "    # Convert extracted timestamps to better-structured form using the function above\n",
+    "    .map(convert_timestamp)\n",
+    "    # Copy document properties to each Document's sub-elements\n",
+    "    .spread_properties([\"path\", \"entity\"])\n",
+    "    # Split elements that are too big to embed\n",
+    "    .split_elements(tokenizer=tokenizer, max_tokens=512)\n",
+    "    # Convert all Elements to Documents\n",
+    "    .explode()\n",
+    "    # Generate a series of hashes to represent each Document. For use with near-duplicate detection\n",
+    "    .sketch()\n",
+    "    # Embed each Document\n",
+    "    .embed(embedder=SentenceTransformerEmbedder(model_name=\"sentence-transformers/all-MiniLM-L6-v2\", batch_size=100))\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Write to a persistent Opensearch Index. Note: You must have a specified opensearch instance running for this to work. For more information on how to set\n",
+    "# one up, refer to https://opensearch.org/docs/latest/install-and-configure/install-opensearch/index/\n",
+    "# - into a specific database (as specified by os_client_args) \n",
+    "# - into a specific index (as specified by index_name)\n",
+    "# - with specific index settings (as specified by index_settings)\n",
+    "os_client_args = {\n",
```

Comment:
Should we have defaults for these to make the config less gnarly? Same for index settings?

Reply:
There are basic defaults and I'm not sure if we want to change it since a lot of code uses the OpenSearch writer. I also think it's useful for users to see the configuration options since that's very important when they want to write to an OpenSearch index